x/concentrated-liquidity/position.go

```diff
@@ -226,3 +228,101 @@ func (k Keeper) getNextPositionIdAndIncrement(ctx sdk.Context) uint64 {
 	k.SetNextPositionId(ctx, nextPositionId+1)
 	return nextPositionId
 }
+
+// fungifyChargedPosition takes in a list of positionIds and combines them into a single position.
+// The previous positions are deleted from state and the new position ID is returned.
+// An error is returned if the caller does not own all the positions, if the positions are all not fully charged, or if the positions are not all in the same pool / tick range.
+func (k Keeper) fungifyChargedPosition(ctx sdk.Context, owner sdk.AccAddress, positionIds []uint64) (uint64, error) {
+	// Check we meet the minimum number of positions to combine.
+	if len(positionIds) <= MinNumPositionsToCombine {
+		return 0, types.PositionQuantityTooLowError{MinNumPositions: MinNumPositionsToCombine, NumPositions: len(positionIds)}
+	}
+
+	// Check that all the positions are in the same pool, tick range, and are fully charged.
+	// Sum the liquidity of all the positions.
+	poolId, lowerTick, upperTick, liquidity, err := k.validatePositionsAndGetTotalLiquidity(ctx, owner, positionIds)
+	if err != nil {
+		return 0, err
+	}
+
+	// The new position's timestamp is the current block time minus the fully charged duration.
+	joinTime := ctx.BlockTime().Add(-types.FullyChargedDuration)
```

Comment:
Are we sure that this doesn't accidentally qualify the position for an additional two weeks of incentives?